src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -608,37 +614,80 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    .rs.fromJSON(.rs.base64decode(encoded))
 })
 
+.rs.addFunction("parseYamlOpt", function(opt)
+{
+    opt <- sub("^#\\|\\s*", "", opt)
+    if (any(grepl(".+=.+", opt))) # R-style chunk options
+    {
+        opts <- paste(opt, collapse=" ")
+        opts <- knitr:::parse_params(opts)
+    }
+    else
+    {
+        opts <- .rs.fromYAML(opt)
+        # parse truthy and falsy style yaml options
+        opts <- lapply(opts, function(value)
+                             {
+                               if (!is.character(value))
+                                 return(value)
+                               if (tolower(value) %in% c("y", "yes", "on"))
+                                 return(TRUE)
+                               if (tolower(value) %in% c("n", "no", "off"))
+                                 return(FALSE)
+                               else
+                                 return(value)
+                             })
+    }
+    return(opts)
+})
+
 .rs.addFunction("evaluateChunkOptions", function(code)
 {
   opts <- list()
 
   Encoding(code) <- "UTF-8"
-  # if several lines of code are passed, operate only on the first 
-  code <- unlist(strsplit(code, "\n", fixed = TRUE))[[1]]
+  # if several lines of code are passed, we expect the first line to contain the standard Rmd chunk options
+  # the next several lines may or may not contain YAML-style chunk options
+  code <- unlist(strsplit(code, "\n", fixed = TRUE))
+  rmdChunkOpts <- code[[1]]
+  yamlChunkOpts <- c()
+
+  for (line in code[-1]) {
+    match <- unlist(regmatches(line, regexec(.rs.reYamlOptChunkBegin(), line)))
+    # there may be variable number of yaml chunk opts, but they should all be at the top of the chunk
+    if (length(match) == 0)
+        break
+    yamlChunkOpts <- c(yamlChunkOpts, match)
+  }
+  opts <- .rs.parseYamlOpt(yamlChunkOpts)
```

Comment:
Should we handle errors here somehow? (I think we at least need a tryCatch() that returns a sensible default on error; not sure if we need to log or report anything to the user)

Reply:
Because we aren't really parsing the options one at a time, I don't think it'll be really possible to parse some options but ignore only the failing ones; however we can just ignore all chunk body options if they fail to parse for any reason

We could do something like 
```
if (length(yamlChunkOpts) > 0)
  {
     opts <- tryCatch(
     .rs.parseYamlOpt(yamlChunkOpts),
     error = function(e) {
        warning("Failed to parse chunk options in body:\n", e)
     })
  }
```

I don't have a good sense of whether those error messages would be helpful, but I think it's probably best to display something to the user so they know the options aren't just being silently ignored (as they are now)